From 61263390183d583574013cbb45f3d3d30b5ab6d6 Mon Sep 17 00:00:00 2001
From: Suresh Gupta <suresh.gupta@xilinx.com>
Date: Wed, 14 Nov 2018 17:03:25 +0530
Subject: [PATCH 0486/1851] v4l: xilinx-multi-scaler: Fix the size of 2nd plane
 for 4:2:2 video formats

commit 0a84d061d52d5a3f015fa0a938e294d1a8848bd5 from
https://github.com/Xilinx/linux-xlnx.git

The size of the 2nd plane for NV12 is half the size
of 1st plane, this patch fix the size.

Signed-off-by: Suresh Gupta <suresh.gupta@xilinx.com>
Reviewed-by: Sandip Kothari <sandipk@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-multi-scaler.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-multi-scaler.c b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
index e500d07aae56..d595afb1f22a 100644
--- a/drivers/media/platform/xilinx/xilinx-multi-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
@@ -1407,6 +1407,15 @@ vidioc_s_fmt(struct xm2msc_chan_ctx *chan_ctx, struct v4l2_format *f)
 		pix->plane_fmt[i].sizeimage = q_data->sizeimage[i];
 	}
 
+	/* Size of 2nd plane of Y_UV10_420 & Y_UV8_420 is half of 1st plane */
+	if (q_data->fmt->xm2msc_fmt == XILINX_M2MSC_FMT_Y_UV10_420 ||
+	    q_data->fmt->xm2msc_fmt == XILINX_M2MSC_FMT_Y_UV8_420) {
+		q_data->sizeimage[1] =
+			q_data->stride * (q_data->height / 2);
+		pix->plane_fmt[1].sizeimage =
+			q_data->stride * (q_data->height / 2);
+	}
+
 	xm2msc_pr_q(chan_ctx->xm2msc_dev->dev, q_data,
 		    chan_ctx->num, f->type, __func__);
 
-- 
2.31.1

